week2/game_2048/class_version/game_template.py

```diff
@@ -0,0 +1,230 @@
+        return self._move_grid_indices
+
+    def generate_move_grid_indices(self):
+        """
+        Returns a dictionary with lists of indices for each row/column
+        in the grid
+        """    
+        return {
+            UP: [(0, index) for index in range(self.get_grid_width())],
+            DOWN: [(self.get_grid_height() - 1, index) for index in range(self.get_grid_width())],
+            RIGHT: [(index, self.get_grid_width() - 1) for index in range(self.get_grid_height())],
+            LEFT: [(index, 0) for index in range(self.get_grid_height())]
+        }
+
+    def get_grid_height(self):
+        """
+        Get the height of the board.
+        """
+        return self._grid_height
+
+    def get_grid_width(self):
+        """
+        Get the width of the board.
+        """
+        return self._grid_width
+
+    def get_grid(self):
+        """
+        Get the grid.
+        """
+        return self._grid
+
+    def set_grid(self, grid):
+        """
+        Set the value of the game grid.
+        """
+        self._grid = grid
+
+    def set_tile(self, row, col, value):
+        """
+        Set the tile at position row, col to have the given value.
+        """
+        self._grid[row][col] = value
+
+    def get_tile(self, row, col):
+        """
+        Return the value of the tile at position row, col.
+        """
+        return self._grid[row][col]
+
+    def reset(self):
+        """
+        Reset the game so the grid is empty except for two
+        initial tiles.
+        """
+        #self.set_grid(self.generate_empty_grid())
+        [self.new_tile() for dummy_num in range(2)]
+
+    def generate_empty_grid(self):
+        """
+        Generates an empty grid of the size
+        _grid_height by _grid_width
+        """
+        return [[ 0 for dummy_col in range(self._grid_width)] for dummy_row in range(self._grid_height)]
+
+    def __str__(self):
+        """
+        Return a string representation of the grid for debugging.
+        """
+        return str(self.get_grid()).replace('],', '],\n')
+
+    def set_direction(self, value):
+        """
+        Sets the direction of the move
+        """
+        self._direction = value
+
+    def get_direction(self):
+        """
+        Returns the direction of the move
+        """
+        return self._direction
+
+    def move(self, direction):
+        """
+        Move all tiles in the given direction and add
+        a new tile if any tiles moved.
+        """
+        self.set_direction(direction)
+
+        if direction in (UP, DOWN):        
+            self.move_tiles(self.get_grid_height())
+        else:
+            self.move_tiles(self.get_grid_width())
+
+    def move_tiles(self, height_width):
+        """
+        Moves tiles to their new location on the grid
+        """
+        new_grid = self.generate_empty_grid()
+        for row in self.generate_index_grid(height_width):
+            self.add_merged_to_grid(row, new_grid)
```

Comment:
Same question here about whether or not to use a list comprehension when not using the values in the array that it generates.